Start a debug guide (#222)
cert-manager/website: content/en/docs/faq/troubleshooting.md

---
title: "Troubleshooting"
linkTitle: "Troubleshooting"
weight: 60
type: "docs"
---

> When troubleshooting cert-manager your best friend is `kubectl describe`, this will give you information on the resources as well as recent events. It it not advised to use the logs as these are quite verbose and only should be looked at if the following steps do not provide help.

Reviewer: It **is** not advised

Author: thx!

> cert-manager consists of multiple custom resources that live inside your Kubernetes cluster, these resources are linked together and are often created by one another. When such an event happens it will reflect in a Kubernetes event, you can see these per namespace using `kubectl get event`, in case of getting them of a specific resource `kubectl describe` will help you a lot.

Reviewer 2: ```suggestion
cert-manager consists of multiple custom resources that live inside your Kubernetes cluster, these resources are linked together and are often created by one another. When such an event happens it will be reflected in a Kubernetes event, you can see these per-namespace using `kubectl get event`, or in the output of `kubectl describe` when looking at a single resource.
```

### 2. Checking the `CertificateRequest`
The `CertificateRequest` resource represents a CSR in cert-manager and passes this CSR on onto the issuer.
You can find the name of the `CertificateRequest` in the `Certificate` event log or using `kubectl get certificaterequest`

To get more info we again run `kubectl describe`:
```console
$ kubectl describe certificaterequest <CertificateRequest name>
API Version:  cert-manager.io/v1alpha3
Kind:         CertificateRequest
Spec:
  Csr: [...]
  Issuer Ref:
    Group:  cert-manager.io
    Kind:   ClusterIssuer
    Name:   letencrypt-production
Status:
  Conditions:
    Last Transition Time:  2020-05-15T21:45:36Z
    Message:               Waiting on certificate issuance from order example-tls-fqtfg-1165244518: "pending"
    Reason:                Pending
    Status:                False
    Type:                  Ready
Events:
  Type    Reason        Age    From          Message
  ----    ------        ----   ----          -------
  Normal  OrderCreated  8m20s  cert-manager  Created Order resource example-tls-fqtfg-1165244518
```

> Here we will see any issuer regarding the Issuer configuration as well as Issuer responses. 

Reviewer 2: ```suggestion
Here we will see any issue regarding the Issuer configuration as well as Issuer responses. 
```
